waku/waku_rln_relay/rln/wrappers.nim

```diff
@@ -353,13 +353,37 @@ type
     lastProcessedBlock*: uint64
     chainId*: uint64
     contractAddress*: string
+    validRoots*: seq[MerkleNode]
 
 proc serialize(metadata: RlnMetadata): seq[byte] =
   ## serializes the metadata
   ## returns the serialized metadata
   return concat(@(metadata.lastProcessedBlock.toBytes()),
                 @(metadata.chainId.toBytes()),
-                @(hexToSeqByte(toLower(metadata.contractAddress))))
+                @(hexToSeqByte(toLower(metadata.contractAddress))),
+                @(uint64(metadata.validRoots.len()).toBytes()),
+                @(serialize(metadata.validRoots)))
+
+type MerkleNodeSeq = seq[MerkleNode]
+
+proc deserialize*(T: type MerkleNodeSeq, merkleNodeByteSeq: seq[byte]): T =
+  ## deserializes a byte seq to a seq of MerkleNodes
+  ## the order of serialization is |merkle_node_len<8>|merkle_node[len]|
+  
+  var roots = newSeq[MerkleNode]()
+  var i = 1'u64
+  let len = uint64.fromBytes(merkleNodeByteSeq[0..7], Endianness.littleEndian)
+  trace "length of valid roots", len
+  var offset = 8'u64
+  while i <= len:
```

Comment:
Minor Q: any reason why not a `for` loop?

Reply:
Thanks, addressed in [c69728f](https://github.com/waku-org/nwaku/pull/1953/commits/c69728f1e0b5f318839622630210afdad979fb9f)